src/stan/lang/generator/generate_transform_inits_method.hpp

```diff
@@ -41,6 +41,8 @@ namespace stan {
         << EOL;
       o << INDENT << "                     std::ostream* pstream__) const {"
         << EOL;
+      o << INDENT2 << "typedef double local_scalar_t__;"
```

Comment:
I don't see what `local_scalar_t__` is doing---this is the only instance in the file.  

I wonder if it's sufficient given that parameters can be involved in the bounds, e.g.,
```
parameters {
  real alpha;
  real<lower = (alpha > 0 ? alpha : 0)> gamma;
}
```

Reply:
this change adds the above typedef as the first statement in the body of the `transform_inits` method.
if an expression used in the lower/upper bounds or location/scale (offset/multiplier) contains a parameter and the generated code requires a call to ` stan::math::promote_scalar`,  then the generated expression will be in terms of `local_scalar_t__` - e.g.:

```
            writer__.scalar_lb_unconstrain((logical_gt(alpha, 0) ? stan::math::promote_scalar<local_scalar_t__>(alpha) : stan::math::promote_scalar<local_scalar_t__>(0) ), gamma);
```

adding the typedef at the beginning of the `transform_inits` method makes this do the right thing.  note that the generated code for the constructor has exactly the same typedef as the first statement.